_posts/2022-04-22-switch-to-java-11.adoc

```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title:  Switching to Java 11/17
+date:   2022-04-22 12:00:00 +0100
+tags: [ community, news ]
+author: vjuranek
+---
+
+As you probably noticed, we have started work on Debezium 2.0.
+One of https://issues.redhat.com/browse/DBZ-3899[the planned changes] for the 2.0 release is https://issues.redhat.com/browse/DBZ-4949[to switch to Java 11 as a baseline].
+While some Java implementations still supports Java 8 
+(e.g. OpenJDK will be https://access.redhat.com/articles/1299013[supported by Red Hat] till 2026), 
+free Oracle JDK 8 already https://www.oracle.com/java/technologies/java-se-support-roadmap.html[has reached end of life]. 
+So it's time to start thinking about using a more recent JDK, and the new major Debezium release is a natural milestone when to do the switch.
+
++++<!-- more -->+++
+
+Since early releases of Debezium 2.0, hopefully from the very first one `Alpha1`, Debezium bits will be compiled to Java 11 byte code.
+Therefore, Java 11 will be required to run Debezium in the next major update.
+Also, if you use any of the Debezium bits in your project, you will have to switch to Java 11.
+
+But wait, what does Java 11/17 in the title mean?
+Is it there just to scare you or we are going to actually switch to Java 17?
+
+<dramatic pause here>
+
+No, we don't want to scare you.
+We are actually planning to switch to Java 17, but only for the test suite.
+This will allow us to use more recent Java features in the tests, while users still can stay on Java 11.
+
+== Implementation
+
+Setting different byte code level for code and tests is pretty easy with Maven, 
```

Comment:
Worth mentioning here that we use the "release" option rather than the legacy "source" and "target" options (see https://www.morling.dev/blog/bytebuffer-and-the-dreaded-nosuchmethoderror/ for the details).